sklearn/metrics/tests/test_classification.py

```diff
@@ -101,6 +102,21 @@ def make_prediction(dataset=None, binary=False):
 ###############################################################################
 # Tests
 
+def test_top_n_accuracy_score():
+    # Test top_n_accuracy_score on three-class prediction task
+    y_true, y_pred, pred_proba = make_prediction(binary=False)
+    # edge case, perfect accuracy
+    assert_equal(top_n_accuracy_score(y_true, pred_proba, n=3), 1.0)
+    # edge case. Should be the same as accuracy_score, but 'probabilities'
+    # for SVM don't always align with class predictions. This happens with
```

Comment:
Do you mean that the class with the highest probability is not necessarily the one getting predicted in `SVC`? Do you have a reference for this? 

It looks like the probabilities from `make_prediction` are never used in `sklearn/metrics/tests/test_classification.py` although they are used in `test_ranking.py`. I'm not sure what was the original motivation for using SVC in this benchmark function (last modified 3 years ago), but if what you are saying is confirmed I don't think that attempting to fix it's output is the right approach. IMO it would be better to fix it in `make_prediction` e.g. by adding an additional parameter allowing to use a different classifier that doesn't have this issues.

Reply:
Why use any model at all and not just create some probabilities?

Reply:
Also, there should be some non-trivial tests that are easy to check "by hand"

Reply:
I was thinking specifically of the sklearn implementation with Platt scaling. The issue is described in the docs [here](http://scikit-learn.org/stable/modules/svm.html) in section 1.4.1.2 for a binary case. @amueller no particular reason to use a model; just trying to take advantage of what was already there - I'll make some simple cases that can be verified by hand.